Lib/test/test_asyncio/test_locks.py

```diff
@@ -928,5 +940,404 @@ def test_release_no_waiters(self):
         self.assertFalse(sem.locked())
 
 
+class BarrierTests(test_utils.TestCase):
+    """
+    Tests for Barrier objects.
+    """
+    def setUp(self):
+        super().setUp()
+        self.loop = self.new_test_loop()
+        self.N = 5
+
+    async def run_coros(self, n, coro):
+        tasks = [self.loop.create_task(coro()) for _ in range(n)]
+        res = await asyncio.gather(*tasks)
+        return res, tasks
+
+    def cancel_coros(self):
+        for task in asyncio.all_tasks(self.loop):
+            task.cancel()
+        test_utils.run_briefly(self.loop)
+
+    def test_repr(self):
+        async def coro():
+            try:
+                i = await barrier.wait()
+            except:
+                ...
+            else:
+                return await asyncio.sleep(1, True)
+
+        self.N = 1011001
+        barrier = asyncio.Barrier(self.N)
+        self.assertTrue("[unset," in repr(barrier))
+        self.assertTrue(f"count:0/{str(self.N)}" in repr(barrier))
+        self.assertTrue(RGX_REPR.match(repr(barrier)))
+        self.assertTrue(repr(barrier).endswith('unset, state:0]>'))
+
+        nb_waits = 3
+        tasks = []
+        for _ in range(nb_waits):
+            tasks.append(self.loop.create_task(coro()))
+        test_utils.run_briefly(self.loop)
+        self.assertTrue("[unset," in repr(barrier))
+        self.assertTrue(RGX_REPR.match(repr(barrier)))
+        self.assertTrue(f"count:{nb_waits}/{self.N}" in repr(barrier))
+        self.assertTrue(repr(barrier).endswith('unset, state:0]>'))
+
+        barrier.reset()
+        self.assertTrue(RGX_REPR.match(repr(barrier)))
+        self.assertTrue("[set," in repr(barrier))
+        self.assertTrue(f"count:{nb_waits}/{self.N}" in repr(barrier))
+        self.assertTrue(repr(barrier).endswith('unset, state:-1]>'))
+        test_utils.run_briefly(self.loop)
+        self.assertTrue(RGX_REPR.match(repr(barrier)))
+        self.assertTrue("[unset," in repr(barrier))
+        self.assertTrue(f"count:0/{self.N}" in repr(barrier))
+        self.assertTrue(repr(barrier).endswith('set, state:0]>'))
+        test_utils.run_briefly(self.loop)
+
+        barrier.abort()
+        test_utils.run_briefly(self.loop)
+        self.assertTrue(RGX_REPR.match(repr(barrier)))
+        self.assertTrue(f"0/{self.N}" in repr(barrier))
+        self.assertTrue(repr(barrier).endswith(':-2]>'))
+
+        self.cancel_coros()
+
+    def test_init(self):
+        self.assertRaises(ValueError, lambda: asyncio.Barrier(0))
+        self.assertRaises(ValueError, lambda: asyncio.Barrier(-4))
+
+    def test_wait_task_once(self):
+        self.N = 1
+        barrier = asyncio.Barrier(self.N)
+        self.assertEqual(barrier.n_waiting, 0)
+        r1 = self.loop.run_until_complete(barrier.wait())
+        self.assertEqual(barrier.n_waiting, 0)
```

Comment:
No, because in this test, Barrier object is created with a ``parties`` parameter to 1. 
So when call ``barrier.wait()``, you never wait, you instantly **drain** 0 task.  So in this case, barrier.n_waiting is always to 0